documentation/content/en/articles/committers-guide/_index.adoc

```diff
@@ -2019,6 +2019,93 @@ git log refs/freebsd/vendor/zlib/1.2.10
 ....
 
 would look at the log for the vendor branch for zlib starting at 1.2.10.
+=== Collaborating with Others
+
+One of the keys to good software development on a project as large as FreeBSD is the ability to collaborate with others before you push your changes to the tree.  The FreeBSD project's git repositories do not, yet, allow user created branches to be pushed to the repository, and therefore if you wish to share your changes with others you must use another mechanism, such as a hosted GitLab or github, in order to share changes in a user generate branch.
+
+The following instructions show how to set up a user generated branch, based on the FreeBSD main branch, and push it to github.
+
+Before you begin, make sure that your local git repo is up to date and has the correct origins set as shown above.
+
+[source,shell]
+````
+% git remote -v
+freebsd https://git.freebsd.org/src.git (fetch)
+freebsd ssh://git@gitrepo.freebsd.org/src.git (push)
+````
+
+The first step is to create a fork of https://github.com/freebsd/freebsd-src[FreeBSD] on github.
+The destination of the fork should be your own, personal, github account (gvnn3 in my case).
+
+Now add a remote that points to your fork:
+....
+% git remote add collab git@github.com:gvnn3/freebsd-src.git
+% git remote -v
+collab	git@github.com:gvnn3/freebsd-src.git (fetch)
+collab	git@github.com:gvnn3/freebsd-src.git (push)
+freebsd	https://git.freebsd.org/src.git (fetch)
+freebsd	ssh://git@gitrepo.freebsd.org/src.git (push)
+....
+With this in place you can create a local branch as shown above.
+
+....
+% git checkout -b gnn-collab
```

Comment:
Hmm, I think to avoid setting up the upstream later on you should branch it off of the collab origin, no?

That is: git remote add github ..., git fetch github (to get the refs), git branch -b myfeature github/main

can you test whether this obsoletes setting up upstream? I also typically push with `git push github HEAD:some_name` or whatever I'm calling that feature branch.

There's too many ways to do anything in git, eh? :)

Reply:
well git branch -b isn't a thing, but I think you mean git checkout -b so I gave this a try:

``
> git checkout -b gnn-feature github/main
Branch 'gnn-feature' set up to track remote branch 'main' from 'github'.
Switched to a new branch 'gnn-feature'

> touch test.c
> git add test.c
> git commit
[gnn-feature 50015e592955] A test file for a test branch
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 test.c
> git push github
fatal: The upstream branch of your current branch does not match
the name of your current branch.  To push to the upstream branch
on the remote, use

    git push github HEAD:main

To push to the branch of the same name on the remote, use

    git push github HEAD
``